asab/web/auth/service.py

```diff
@@ -544,15 +548,17 @@ def _validate_wrapper_installation(self):
 				# Authorization wrapper not installed here
 				continue
 
-			# Ensure the wrappers are applied in the correct order
-			try:
-				tenant_wrapper_idx = web_container.WebApp.on_startup.index(set_up_tenant_web_wrapper)
-			except ValueError:
-				# Tenant wrapper not installed here
+			if tenant_service is None:
+				# Without tenant service there are no tenant web wrappers
 				continue
 
-			if auth_wrapper_idx > tenant_wrapper_idx:
-				raise Exception("TenantService.install() must be called before AuthService.install()")
+			# Ensure the wrappers are applied in the correct order
+			tenant_wrapper_idx = tenant_service.get_web_wrapper_position(web_container)
+			if tenant_wrapper_idx is not None and auth_wrapper_idx > tenant_wrapper_idx:
+				L.error(
+					"TenantService.install(web_container) must be called before AuthService.install(web_container). "
+					"Otherwise authorization will not work properly."
```

Comment:
This message may be misleading, b/c people are not calling `install()` explicitly, right?

Reply:
this condition should only ever be true if someone calls install() explicitly.

if they rely on the automatic installation instead, this error should never occur.